cachi2/core/package_managers/yarn/project.py

```diff
@@ -4,10 +4,21 @@
 It also provides basic utility functions. The main logic to resolve and prefetch the dependencies
 should be implemented in other modules.
 """
+import json
+import logging
 from typing import Any, NamedTuple, Optional
 
+import yaml
+
+from cachi2.core.errors import PackageRejected
 from cachi2.core.rooted_path import RootedPath
 
+log = logging.getLogger(__name__)
+
+
+DEFAULT_CACHE_FOLDER = "./.yarn/cache"
```

Comment:
Good point, for the outside-RHTAP use case a user could indeed be setting `YARN_` environment variables

Even the path to .yarnrc.yml itself is configurable via env var (but this would _only_ work outside RHTAP, where the user can't set env vars) https://v3.yarnpkg.com/configuration/yarnrc#rcFilename

Reply:
> Even the path to .yarnrc.yml itself is configurable via env var (but this would _only_ work outside RHTAP, where the user can't set env vars I think) https://v3.yarnpkg.com/configuration/yarnrc#rcFilename

Or maybe they can, I never know which part of the definition the user can edit or not https://tekton.dev/docs/pipelines/tasks/#specifying-a-step-template

Reply:
> Good point, for the outside-RHTAP use case a user could indeed be setting `YARN_` environment variables
> 
> Even the path to .yarnrc.yml itself is configurable via env var (but this would _only_ work outside RHTAP, where the user can't set env vars I think) https://v3.yarnpkg.com/configuration/yarnrc#rcFilename

Aand as it turns out, in this particular case `YARN_CACHE_FOLDER` is a valid env variable.

Reply:
If we always set the `env` parameter when [calling](https://github.com/containerbuildsystem/cachi2/blob/main/cachi2/core/package_managers/yarn/utils.py#L14) the subprocess, we will override any environment variable that is set in the user's env. So I think the solution here is being sure this parameter is at least set to an empty dict.

Reply:
If we do want to expand our use case to also consider the environment variables that are already set, then we probably will need to have a separate parsing of `YARN_*` environment variables and always consider precedence for them instead of what's in yarnrc (see the early work in what we will need to support [here](https://github.com/brunoapimentel/cachi2/blob/set-yarn-config/cachi2/core/package_managers/yarn/project.py#L48-L115)).

I think it's definitely a valid use case, but we should probably handle that in a separate PR. In any case, the variable here is meant to represent the fallback value for when `cacheFolder` is not set in .yarnrc, so I think it is still correct.

Reply:
Yeah, we either need to document it as unsupported or support it.

Reply:
I'll file a story so we can discuss what's the best approach.